components/workspacekit/pkg/seccomp/notify.go

```diff
@@ -284,25 +292,42 @@ func (h *InWorkspaceHandler) Mount(req *libseccomp.ScmpNotifReq) (val uint64, er
 			}
 		}
 
-		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-		defer cancel()
-		iws, err := h.Daemon(ctx)
-		if err != nil {
-			log.WithField("target", target).WithField("dest", dest).WithField("mode", stat.Mode()).WithError(err).Errorf("cannot get IWS client to mount %s", filesystem)
-			return Errno(unix.EFAULT)
-		}
-		defer iws.Close()
+		wait := iwsBackoffInitialWait
+		for i := 0; i < iwsBackoffSteps; i++ {
+			err = func() error {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				iws, err := h.Daemon(ctx)
+				if err != nil {
+					log.WithField("target", target).WithField("dest", dest).WithField("mode", stat.Mode()).WithError(err).Errorf("cannot get IWS client to mount %s", filesystem)
+					return err
+				}
+				defer iws.Close()
 
-		call := iws.MountProc
-		if filesystem == "sysfs" {
-			call = iws.MountSysfs
+				call := iws.MountProc
+				if filesystem == "sysfs" {
+					call = iws.MountSysfs
+				}
+				_, err = call(ctx, &daemonapi.MountProcRequest{
+					Target: dest,
+					Pid:    int64(req.Pid),
+				})
+				if err != nil {
+					log.WithField("target", dest).WithError(err).Errorf("cannot mount %s", filesystem)
+					return err
+				}
+				return nil
+			}()
+			if err != nil {
+				time.Sleep(wait)
+				wait = wait * iwsBackoffFactor
+				if wait > iwsBackoffMaxWait {
+					wait = iwsBackoffMaxWait
+				}
+			}
```

Comment:
Can we use a library to do the exponential backoff? [[1](https://github.com/gitpod-io/gitpod/blob/4c9a2b11900dee4196a9291cd94719e4eba0ad11/components/ws-manager-mk2/controllers/workspace_controller.go#L440)][[2](https://pkg.go.dev/k8s.io/apimachinery@v0.26.2/pkg/util/wait#ExponentialBackoffWithContext)]

I'm not super comfortable with `workspacekit`, and will schedule some time to talk through synchronously Thursday. :pray: